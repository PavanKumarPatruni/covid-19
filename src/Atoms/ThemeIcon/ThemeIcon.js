import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';

import { Colors } from '../../utils/contants';

const Themeicon = forwardRef(({ size, darkMode }, ref) => {
  return darkMode ? (
    <svg
      ref={ref}
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill="none"
      stroke={Colors.YELLOW}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  ) : (
    <svg
      ref={ref}
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      viewBox="0 0 24 24"
      fill={Colors.VOILET}
      stroke={Colors.VOILET}
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    >
      <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
    </svg>
  );
});

Themeicon.propTypes = {
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  darkMode: PropTypes.bool,
};

Themeicon.defaultProps = {
  size: 24,
  darkMode: true,
};

Themeicon.displayName = 'ArrowUp';

export default Themeicon;
