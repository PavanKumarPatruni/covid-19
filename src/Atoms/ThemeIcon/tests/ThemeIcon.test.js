import React from 'react';
import { shallow, mount } from 'enzyme';

import ThemeIcon from '../ThemeIcon';

describe('ThemeIcon component', () => {
  const props = {
    size: 24,
    darkMode: false,
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<ThemeIcon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('match snapshot correctly', () => {
    const wrapper = mount(<ThemeIcon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    props.darkMode = true;
    const wrapper = shallow(<ThemeIcon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('match snapshot correctly', () => {
    props.darkMode = true;
    const wrapper = mount(<ThemeIcon {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
