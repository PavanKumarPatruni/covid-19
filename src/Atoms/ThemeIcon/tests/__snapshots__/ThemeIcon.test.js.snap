// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeIcon component match snapshot correctly 1`] = `
<svg
  fill="#201c3a"
  height={24}
  stroke="#201c3a"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
  />
</svg>
`;

exports[`ThemeIcon component match snapshot correctly 2`] = `
<ArrowUp
  darkMode={false}
  size={24}
>
  <svg
    fill="#201c3a"
    height={24}
    stroke="#201c3a"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"
    />
  </svg>
</ArrowUp>
`;

exports[`ThemeIcon component match snapshot correctly 3`] = `
<svg
  fill="none"
  height={24}
  stroke="#ffc107"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth="2"
  viewBox="0 0 24 24"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="12"
    cy="12"
    r="5"
  />
  <line
    x1="12"
    x2="12"
    y1="1"
    y2="3"
  />
  <line
    x1="12"
    x2="12"
    y1="21"
    y2="23"
  />
  <line
    x1="4.22"
    x2="5.64"
    y1="4.22"
    y2="5.64"
  />
  <line
    x1="18.36"
    x2="19.78"
    y1="18.36"
    y2="19.78"
  />
  <line
    x1="1"
    x2="3"
    y1="12"
    y2="12"
  />
  <line
    x1="21"
    x2="23"
    y1="12"
    y2="12"
  />
  <line
    x1="4.22"
    x2="5.64"
    y1="19.78"
    y2="18.36"
  />
  <line
    x1="18.36"
    x2="19.78"
    y1="5.64"
    y2="4.22"
  />
</svg>
`;

exports[`ThemeIcon component match snapshot correctly 4`] = `
<ArrowUp
  darkMode={true}
  size={24}
>
  <svg
    fill="none"
    height={24}
    stroke="#ffc107"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="5"
    />
    <line
      x1="12"
      x2="12"
      y1="1"
      y2="3"
    />
    <line
      x1="12"
      x2="12"
      y1="21"
      y2="23"
    />
    <line
      x1="4.22"
      x2="5.64"
      y1="4.22"
      y2="5.64"
    />
    <line
      x1="18.36"
      x2="19.78"
      y1="18.36"
      y2="19.78"
    />
    <line
      x1="1"
      x2="3"
      y1="12"
      y2="12"
    />
    <line
      x1="21"
      x2="23"
      y1="12"
      y2="12"
    />
    <line
      x1="4.22"
      x2="5.64"
      y1="19.78"
      y2="18.36"
    />
    <line
      x1="18.36"
      x2="19.78"
      y1="5.64"
      y2="4.22"
    />
  </svg>
</ArrowUp>
`;
